pkg/encoding/encoding.go

```diff
@@ -17,6 +17,9 @@ func Decode(data []byte) (*spec.App, error) {
 		return nil, errors.Wrap(err, "could not unmarshal into internal struct")
 	}
 	log.Debugf("object unmrashalled: %#v\n", app)
+	if err := sanifyApp(&app); err != nil {
```

Comment:
can we name it like `populateApp` or maybe `resolveApp` that it could be point where also all the shortcuts are handled.
Than we can always expect that after this function is called all the kubernets objects in spec are fully populated with all required fields and transformer just needs to reorganize it back into Kubernetes primitives